drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua

```diff
@@ -195,74 +197,81 @@ local function configure(driver, device)
-    profile_name = profile_name .. "-pm25"
+    level_indication_support = level_indication_support .. "-pm25"
   end
   if #pm10_level_eps > 0 then
-    profile_name = profile_name .. "-pm10"
+    level_indication_support = level_indication_support .. "-pm10"
   end
   if #radon_level_eps > 0 then
-    profile_name = profile_name .. "-radon"
+    level_indication_support = level_indication_support .. "-radon"
   end
   if #tvoc_level_eps > 0 then
-    profile_name = profile_name .. "-tvoc"
+    level_indication_support = level_indication_support .. "-tvoc"
   end
-  if #co_level_eps > 0 or #co2_level_eps > 0 or #no2_level_eps > 0 or #ozone_level_eps > 0 or
-      #formaldehyde_level_eps > 0 or #pm1_level_eps > 0 or #pm2_5_level_eps > 0 or #pm10_level_eps > 0 or
-      #radon_level_eps > 0 or #tvoc_level_eps > 0 then
-    profile_name = profile_name .. "-level"
+  if #co_level_eps > 0 and #co2_level_eps > 0 and #no2_level_eps > 0 and #ozone_level_eps > 0 and
+      #formaldehyde_level_eps > 0 and #pm1_level_eps > 0 and #pm2_5_level_eps > 0 and #pm10_level_eps > 0 and
+      #radon_level_eps > 0 and #tvoc_level_eps > 0 then
+    level_indication_support = "-all"
   end
+  if not (level_indication_support == "") then
+    profile_name = profile_name .. level_indication_support .. "-level"
+  end
+
   if #co_meas_eps > 0 then
-    profile_name = profile_name .. "-co"
+    numeric_measurement_support = numeric_measurement_support .. "-co"
   end
   if #co2_meas_eps > 0 then
-    profile_name = profile_name .. "-co2"
+    numeric_measurement_support = numeric_measurement_support .. "-co2"
   end
   if #no2_meas_eps > 0 then
-    profile_name = profile_name .. "-no2"
+    numeric_measurement_support = numeric_measurement_support .. "-no2"
   end
   if #ozone_meas_eps > 0 then
-    profile_name = profile_name .. "-ozone"
+    numeric_measurement_support = numeric_measurement_support .. "-ozone"
   end
   if #formaldehyde_meas_eps > 0 then
-    profile_name = profile_name .. "-formaldehyde"
+    numeric_measurement_support = numeric_measurement_support .. "-formaldehyde"
   end
   if #pm1_meas_eps > 0 then
-    profile_name = profile_name .. "-pm1"
+    numeric_measurement_support = numeric_measurement_support .. "-pm1"
   end
   if #pm2_5_meas_eps > 0 then
-    profile_name = profile_name .. "-pm25"
+    numeric_measurement_support = numeric_measurement_support .. "-pm25"
   end
   if #pm10_meas_eps > 0 then
-    profile_name = profile_name .. "-pm10"
+    numeric_measurement_support = numeric_measurement_support .. "-pm10"
   end
   if #radon_meas_eps > 0 then
-    profile_name = profile_name .. "-radon"
+    numeric_measurement_support = numeric_measurement_support .. "-radon"
   end
   if #tvoc_meas_eps > 0 then
-    profile_name = profile_name .. "-tvoc"
+    numeric_measurement_support = numeric_measurement_support .. "-tvoc"
+  end
+  if #co_meas_eps > 0 and #co2_meas_eps > 0 and #no2_meas_eps > 0 and #ozone_meas_eps > 0 and
```

Comment:
I would recommend to just explictly check if the constructed profile name is  `air-quality-sensor-temp-humidity-co-co2-no2-ozone-formaldehyde-pm1-pm25-pm10-radon-tvoc-level-co-co2-no2-ozone-formaldehyde-pm1-pm25-pm10-radon-tvoc-meas` and then just set that to `air-quality-sensor-all` and do the same for the all meas profile name. We essentially already have all the information we need encoded in the profile name, so we don't need to check for endpoints again.

Also please leave a comment explaining why we need to do this because of the profile name character limit

Reply:
Made this change in 8b445ac741be1ab4257d5ce16d4fbdd4d0d0d0cf and added comment in ec1916d48ca32d021c5e01623ea5357ca4b2bbec.